consensus/fsm_handler.go

```diff
@@ -0,0 +1,111 @@
+		return m.HandleBootstrapped(msg)
+
+	case coreTypes.StateMachineState_Consensus_Unsynched:
+		return m.HandleUnsynched(msg)
+
+	case coreTypes.StateMachineState_Consensus_SyncMode:
+		return m.HandleSyncMode(msg)
+
+	case coreTypes.StateMachineState_Consensus_Synced:
+		return m.HandleSynced(msg)
+
+	case coreTypes.StateMachineState_Consensus_Pacemaker:
+		return m.HandlePacemaker(msg)
+	default:
+		m.logger.Warn().Msg("Consensus module not handling this event")
+
+	}
+
+	return nil
+}
+
+// HandleBootstrapped handles FSM event P2P_IsBootstrapped, and P2P_Bootstrapped is the destination state.
+// Bootrstapped mode is when the node (validator or non-validator) is first coming online.
+// This is a transition mode from node bootstrapping to a node being out-of-sync.
+func (m *consensusModule) HandleBootstrapped(msg *messaging.StateMachineTransitionEvent) error {
+	m.logger.Debug().Msg("FSM is in bootstrapped state, so it is out of sync, and transitions to unsynched mode")
+	return m.GetBus().GetStateMachineModule().SendEvent(coreTypes.StateMachineEvent_Consensus_IsUnsynched)
+}
+
+// HandleUnsynched handles FSM event Consensus_IsUnsynched, and Unsynched is the destination state.
+// In Unsynched mode node (validator or non-validator) is out of sync with the rest of the network.
+// This mode is a transition mode from the node being up-to-date (i.e. Pacemaker mode, Synched mode) with the latest network height to being out-of-sync.
+// As soon as node transitions to this mode, it will transition to the sync mode.
+func (m *consensusModule) HandleUnsynched(msg *messaging.StateMachineTransitionEvent) error {
+	m.logger.Debug().Msg("FSM is in Unsyched state, as node is out of sync sending syncmode event to start syncing")
+
+	return m.GetBus().GetStateMachineModule().SendEvent(coreTypes.StateMachineEvent_Consensus_IsSyncing)
+}
+
+// HandleSyncMode handles FSM event Consensus_IsSyncing, and SyncMode is the destination state.
+// In Sync mode node (validator or non-validator) starts syncing with the rest of the network.
+func (m *consensusModule) HandleSyncMode(msg *messaging.StateMachineTransitionEvent) error {
+	m.logger.Debug().Msg("FSM is in Sync Mode, starting syncing...")
+
+	return m.stateSync.StartSynching()
+}
+
+// HandleSynced handles FSM event IsSynchedNonValidator, and Synced is the destination state.
+// Currently, FSM never transition to this state and a non-validator node always stays in syncmode.
+// CONSIDER: when a non-validator sync is implemented, maybe there is a case that requires transitioning to this state.
+// TODO: Add check that this never happens when IsValidator() is false, i.e. node is not validator.
+func (m *consensusModule) HandleSynced(msg *messaging.StateMachineTransitionEvent) error {
+	m.logger.Debug().Msg("FSM of non-validator node is in Synced mode")
+	return nil
+}
+
+// HandlePacemaker handles FSM event IsSynchedValidator, and Pacemaker is the destination state.
```

Comment:
`Synched` and `Sync` are correct. `Synch` is not.

Leaving one comment here (where there is no delta), but just FYI.